tests/core/test_nlg.py

```diff
@@ -328,3 +331,271 @@ def test_nlg_fill_response_quick_replies(
+    ),
+    (
+        ("test", "A", "test_other", "B", "example AB"),
+        ("test", "C", "test_other", "D", "example CD"),
+    ),
+)
+async def test_nlg_with_multiple_constraints(
+    first_slot, first_slot_value, second_slot, second_slot_value, response_variation
+):
+    responses = {
+        "utter_action": [
+            {
+                "text": "example AB",
+                "condition": [
+                    {"type": "slot", "name": "test", "value": "A"},
+                    {"type": "slot", "name": "test_other", "value": "B"},
+                ],
+            },
+            {
+                "text": "example CD",
+                "condition": [
+                    {"type": "slot", "name": "test", "value": "C"},
+                    {"type": "slot", "name": "test_other", "value": "D"},
+                ],
+            },
+        ]
+    }
+    t = TemplatedNaturalLanguageGenerator(responses=responses)
+
+    first_slot = Slot(
+        name=first_slot, initial_value=first_slot_value, influence_conversation=False
+    )
+    second_slot = Slot(
+        name=second_slot, initial_value=second_slot_value, influence_conversation=False
+    )
+    tracker = DialogueStateTracker(
+        sender_id="test_nlg_multiple", slots=[first_slot, second_slot]
+    )
+
+    r = await t.generate(
+        utter_action="utter_action", tracker=tracker, output_channel=""
+    )
+    assert r.get("text") == response_variation
+
+
+@pytest.mark.parametrize(
+    ("slot_name", "slot_value", "response_variation"),
+    (("test", "A", "example one A"), ("test", "B", "example two B")),
+)
+async def test_nlg_conditional_response_variations_with_interpolated_slots(
+    slot_name, slot_value, response_variation
+):
+    responses = {
+        "utter_action": [
+            {
+                "text": "example one {test}",
+                "condition": [{"type": "slot", "name": "test", "value": "A"}],
+            },
+            {
+                "text": "example two {test}",
+                "condition": [{"type": "slot", "name": "test", "value": "B"},],
+            },
+        ]
+    }
+    t = TemplatedNaturalLanguageGenerator(responses=responses)
+    slot = Slot(name=slot_name, initial_value=slot_value, influence_conversation=False)
+    tracker = DialogueStateTracker(sender_id="nlg_interpolated", slots=[slot])
+
+    r = await t.generate(
+        utter_action="utter_action", tracker=tracker, output_channel=""
+    )
+    assert r.get("text") == response_variation
+
+
+@pytest.mark.parametrize(
+    ("slot_name", "slot_value", "bot_message"),
+    (
+        (
+            "can_withdraw",
+            False,
+            "You are not allowed to withdraw any amounts. Please check permission.",
+        ),
+        (
+            "account_type",
+            "secondary",
+            "Withdrawal was sent for approval to primary account holder.",
+        ),
+    ),
+)
+async def test_nlg_conditional_response_variations_with_yaml_single_condition(
```

Comment:
the yaml tests feel a bit duplicate to me. What do they test what your previous tests don't test? You could also write the responses as yaml string within this module and use it in your tests by doing something like `Domain.from_yaml` -> `Domain.responses`

Reply:
I have received feedback before that I should always test importing from file to make sure everything works as expected in an almost 'real-life' scenario - I guess in the first tests I used a `dict `directly because I wanted to focus on the new implementation rather to bypass any yaml validation errors. 
I could remove anything that appears as a duplicate and just keep the yaml ones.

Reply:
> I have received feedback before that I should always test importing from file to make sure everything works as expected in an almost 'real-life' scenario

That's good feedback 👍🏻 Especially if you think of things like schema validation.

> I guess in the first tests I used a dict directly because I wanted to focus on the new implementation rather to bypass any yaml validation errors.

fair 👍🏻 

> I could remove anything that appears as a duplicate and just keep the yaml ones.

I'd slightly prefer this as I also think the YAML syntax is more intuitive to read / understand when reading the code as well. Not worth to put too much work into this though - you could also just keep the existing tests and maybe reduce the YAML ones a bit and rather treat them as some sort of integration test.